### PR TITLE
viewBinding - add view binding in project

### PR DESCRIPTION
**Pull request by polinas888**

### CODE REVIEW

File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt [hunk 1]
```diff
@@ -26,15 +28,16 @@ class WeekWeatherFragment : Fragment(), WeatherForecastAdapter.ViewHolder.OnItem
...
+        return view
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        weatherForecastRecyclerView = view.findViewById(com.example.perfectweatherallyear.R.id.weatherForecastRecyclerView)
-        weatherForecastRecyclerView.adapter = WeatherForecastAdapter(dayOfWeekList, weatherPerWeekList, this)
-        weatherForecastRecyclerView.layoutManager = LinearLayoutManager(requireContext())
+        binding.weatherForecastRecyclerView.adapter = WeatherForecastAdapter(dayOfWeekList, weatherPerWeekList, this)
+        binding.weatherForecastRecyclerView.layoutManager = LinearLayoutManager(requireContext())
```
Comment (EVIL-SnaKe) [inline app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt:40]: use scope function with(binding) { }

File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeatherForecastAdapter.kt [hunk 2]
```diff
@@ -30,14 +29,13 @@ class WeatherForecastAdapter(private val dayOfWeekList: List<String>, private va
...
-            return ViewHolder(view, mOnItemListener)
+            val binding = WeatherRowItemBinding.inflate(LayoutInflater.from(viewGroup.context), viewGroup, false)
+            return ViewHolder(binding, mOnItemListener)
         }
 
         override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
-            viewHolder.dayOfWeekTextView.text = dayOfWeekList[position]
-            viewHolder.maxMinTemperatureTextView.text = weatherList[position].temperature
+            viewHolder.binding.dayOfWeekTextView.text = dayOfWeekList[position]
+            viewHolder.binding.maxMinTempretureTextView.text = weatherList[position].temperature
```
Comment (EVIL-SnaKe) [inline app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeatherForecastAdapter.kt:38]: Use scope function to viewHolder

File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeatherForecastAdapter.kt [hunk 3]
```diff
@@ -6,15 +6,14 @@ import android.view.ViewGroup
...
 import com.example.perfectweatherallyear.model.DayWeather
 
 class WeatherForecastAdapter(private val dayOfWeekList: List<String>, private val weatherList: List<DayWeather>, onItemListener: ViewHolder.OnItemListener) :
         RecyclerView.Adapter<WeatherForecastAdapter.ViewHolder>() {
     private val mOnItemListener: ViewHolder.OnItemListener = onItemListener
 
-    class ViewHolder(view: View, private val onItemListener: OnItemListener) : RecyclerView.ViewHolder(view), View.OnClickListener {
-        val dayOfWeekTextView: TextView = view.findViewById(R.id.dayOfWeekTextView)
-        val maxMinTemperatureTextView: TextView = view.findViewById(R.id.maxMinTempretureTextView)
+    class ViewHolder(val binding: WeatherRowItemBinding, private val onItemListener: OnItemListener) : RecyclerView.ViewHolder(binding.root), View.OnClickListener {
```
Comment (EVIL-SnaKe) [inline app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeatherForecastAdapter.kt:16]: Move this class to the bottom of the file

File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeatherForecastAdapter.kt [hunk 4]
```diff
@@ -6,15 +6,14 @@ import android.view.ViewGroup
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
 import com.example.perfectweatherallyear.R
+import com.example.perfectweatherallyear.databinding.WeatherRowItemBinding
 import com.example.perfectweatherallyear.model.DayWeather
 
 class WeatherForecastAdapter(private val dayOfWeekList: List<String>, private val weatherList: List<DayWeather>, onItemListener: ViewHolder.OnItemListener) :
```
Comment (EVIL-SnaKe) [inline app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeatherForecastAdapter.kt:12]: Change constructor to receive one data argument: Map<String, DayWeather>

**Review by EVIL-SnaKe — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt [hunk 5]
```diff
@@ -18,34 +18,35 @@ class WeekWeatherFragment : Fragment(), WeatherForecastAdapter.ViewHolder.OnItem
...
+        }
     }
 
     override fun onItemClick(position: Int) {
         val fragment = DetailWeatherFragment()
 
         val args = Bundle()
-        val dayWeather = DayWeather(temperature = weatherPerWeekList[position].temperature,
-            precipitation = weatherPerWeekList[position].precipitation, wind = weatherPerWeekList[position].wind)
+        val currentDayWeather = weekWeatherMap.get(weekWeatherMap.keys.toTypedArray()[position])
```
Comment (EVIL-SnaKe) [inline app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt:46]:
```suggestion
val currentDayWeather = weekWeatherMap[weekWeatherMap.keys.toTypedArray()[position]]
```

File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt [hunk 6]
```diff
@@ -18,34 +18,35 @@ class WeekWeatherFragment : Fragment(), WeatherForecastAdapter.ViewHolder.OnItem
...
+        val dayWeather = currentDayWeather?.let { DayWeather(temperature = it.temperature,
```
Comment (EVIL-SnaKe) [inline app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt:47]: Looks like object dayWeather is redundant. You can use already existed val currentDayWeather.

File: app/src/main/java/com/example/perfectweatherallyear/MainActivity.kt [hunk 7]
```diff
@@ -11,7 +11,6 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
-        val view = binding.root
-        setContentView(view)
+        setContentView(binding.root)
```
Comment (polinas888) [inline app/src/main/java/com/example/perfectweatherallyear/MainActivity.kt:14]: gghj

**Pull request merged**